Source/Particles/ShapeFactors.H

```diff
@@ -23,170 +23,100 @@ struct Compute_shape_factor
 {
     template< typename T >
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-    int operator()(T* const /*sx*/, T /*xint*/) const { return 0; }
-};
-
-/**
- *  Compute shape factor and return index of leftmost cell where
- *  particle writes.
- *  Specialization for order 0
- */
-template <>
-struct Compute_shape_factor< 0 >
-{
-    template< typename T >
-    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-    int operator()(T* const sx, T xmid) const
+    int operator()(
+        T* const sx,
+        T xmid) const
     {
-        const auto j = static_cast<int>(xmid + T(0.5));
-        sx[0] = T(1.0);
-        return j;
+        if constexpr (depos_order == 0){
+            const auto j = static_cast<int>(xmid + T(0.5));
+            sx[0] = T(1.0);
+            return j;
+        }
+        else if constexpr (depos_order == 1){
+            const auto j = static_cast<int>(xmid);
+            const T xint = xmid - T(j);
+            sx[0] = T(1.0) - xint;
+            sx[1] = xint;
+            return j;
+        }
+        else if constexpr (depos_order == 2){
+            const auto j = static_cast<int>(xmid + T(0.5));
+            const T xint = xmid - T(j);
+            sx[0] = T(0.5)*(T(0.5) - xint)*(T(0.5) - xint);
+            sx[1] = T(0.75) - xint*xint;
+            sx[2] = T(0.5)*(T(0.5) + xint)*(T(0.5) + xint);
+            // index of the leftmost cell where particle deposits
+            return j-1;
+        }
+        else if constexpr (depos_order == 3){
+            const auto j = static_cast<int>(xmid);
+            const T xint = xmid - T(j);
+            sx[0] = (T(1.0))/(T(6.0))*(T(1.0) - xint)*(T(1.0) - xint)*(T(1.0) - xint);
+            sx[1] = (T(2.0))/(T(3.0)) - xint*xint*(T(1.0) - xint/(T(2.0)));
+            sx[2] = (T(2.0))/(T(3.0)) - (T(1.0) - xint)*(T(1.0) - xint)*(T(1.0) - T(0.5)*(T(1.0) - xint));
+            sx[3] = (T(1.0))/(T(6.0))*xint*xint*xint;
+            // index of the leftmost cell where particle deposits
+            return j-1;
+        }
+        else{
+            static_assert(false, "Unknown particle shape selected in Compute_shape_factor");
```

Comment:
Haha, `static_assert` in `if constexpr - else` is a non-trivial issue it seems:
https://stackoverflow.com/questions/38304847/constexpr-if-and-static-assert
```suggestion
            amrex::Abort("Unknown particle shape selected in Compute_shape_factor");
```